src/algebra/order/absolute_value.lean

```diff
@@ -37,12 +38,26 @@ section ordered_semiring
 
 variables {R S : Type*} [semiring R] [ordered_semiring S] (abv : absolute_value R S)
 
-instance mul_hom_class : mul_hom_class (absolute_value R S) R S :=
+instance zero_hom_class : zero_hom_class (absolute_value R S) R S :=
 { coe := λ f, f.to_fun,
   coe_injective' := λ f g h, by { obtain ⟨⟨_, _⟩, _⟩ := f, obtain ⟨⟨_, _⟩, _⟩ := g, congr' },
-  map_mul := λ f, f.map_mul' }
+  map_zero := λ f, (f.eq_zero' _).2 rfl }
+
+instance mul_hom_class : mul_hom_class (absolute_value R S) R S :=
```

Comment:
As @Vierkantor comments below `for add_group_seminorm_class.to_zero_hom_class`, wouldn't it be good to make `coe` explicit in this and the following instances?

Reply:
Agreed, copying the data carrying fields is helpful.